source/windows/shared_library.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* In theory clang-format doesn't get run on this, but someday maybe it will*/
+/* clang-format off */
+#include <Windows.h>
+#include <libloaderapi.h>
+/* clang-format on */
+
+#include <aws/io/shared_library.h>
+
+#include <aws/io/logging.h>
+
+static const char *s_null = "<NULL>";
+static const char *s_unknown_error = "<Unknown>";
+
+int aws_shared_library_init(struct aws_shared_library *library, const char *library_path) {
+    AWS_ZERO_STRUCT(*library);
+
+    library->library_handle = LoadLibrary(library_path);
+    if (library->library_handle == NULL) {
+        DWORD ec = GetLastError();
```

Comment:
You can convert this to a string using FormatMessage()

Reply:
I was really reluctant to start using that for windows errors.  We don't use it anywhere else in the platform-specific logic and it's doing windows-internal allocation underneath and I didn't want to be the first =/